days/src/day4/Day4.kt

```diff
@@ -0,0 +1,62 @@
+package day4
+
+import Day
+import java.io.File
+
+class Day4(private val passports: List<Map<String, String>>) : Day {
+    enum class PassportField {
+        BYR, IYR, EYR, HGT, HCL, ECL, PID;//, CID
+
+        override fun toString() = super.toString().toLowerCase()
+        private val _eyeColors = listOf("amb", "blu", "brn", "gry", "grn", "hzl", "oth")
+        fun validate(value: String) = when (this) {
+            BYR -> value.toInt() in 1920..2002
+            IYR -> value.toInt() in 2010..2020
+            EYR -> value.toInt() in 2020..2030
+            HGT -> "(\\d+)(in|cm)".toRegex().matchEntire(value)?.let {
+                when (it.groupValues[2]) {
+                    "in" -> it.groupValues[1].toInt() in 59..76
+                    "cm" -> it.groupValues[1].toInt() in 150..193
+                    else -> false
+                }
+            } ?: false
+            ECL -> value in _eyeColors
+            HCL -> "#[0-9a-f]{6}".toRegex().matches(value)
+            PID -> "\\d{9}".toRegex().matches(value)
```

Comment:
The regex usage here and in height are good. My height validator was pretty long because of the check for the in/cm and then the number value being in range.

The in range functionality in Kotlin is really nice too, I don't think I can construct anything similar in JS :(